spyder/plugins/projects/widgets/main_widget.py

```diff
@@ -598,6 +600,151 @@ def show_widget(self):
+        filename = item.get_data()
+        self.sig_open_file_requested.emit(filename)
+        self.get_plugin()._switcher.hide()
+
+    def handle_switcher_results(self, search_text, items_data):
+        """
+        Handle user typing in switcher to filter result.
+        Load switcher results when a search text is typed for projects.
+        Parameters
+        ----------
+        text: str
+            The current search text in the switcher dialog box.
+        items_data: list
+            List of items shown in the switcher.
+        """
+        switcher = self.get_plugin()._switcher
+        paths = self._execute_fzf_subprocess(search_text)
+
+        for sw_path in items_data:
+            if (sw_path in paths):
+                paths.remove(sw_path)
+
+        is_unsaved = [False] * len(paths)
+        short_paths = shorten_paths(paths, is_unsaved)
+        section = self.get_title()
+
+        for i, (path, short_path) in enumerate(zip(paths, short_paths)):
+            title = osp.basename(path)
+            icon = get_file_icon(path)
+            description = osp.dirname(path).lower()
+            if len(path) > 75:
+                description = short_path
+            is_last_item = (i+1 == len(paths))
+
+            switcher.add_item(
+                title=title,
+                description=description,
+                icon=icon,
+                section=section,
+                data=path,
+                last_item=is_last_item,
+                score=100
+            )
+
+    # fzf helper method
+    def _execute_fzf_subprocess(self, search_text=""):
+        """
+        Execute fzf subprocess to get the list of files in the current
+        project filtering by the search_text.
+        Parameters
+        ----------
+        search_text: str
+            The current search text in the switcher dialog box.
+        """
+        project_path = self.get_active_project_path()
+        if project_path is None:
+            return []
+        # command = fzf --filter <search_str>
+        cmd_list = ["fzf", "--filter", search_text]
+        shell = False
+        env = os.environ.copy()
+        startupinfo = subprocess.STARTUPINFO()
+        try:
+            out = subprocess.check_output(cmd_list, cwd=project_path,
+                                          shell=shell, env=env,
+                                          startupinfo=startupinfo,
+                                          stderr=subprocess.STDOUT)
+            relative_path_list = out.decode('UTF-8').strip().split("\n")
+            # List of tuples with the absolute path
+            result_list = [
+                osp.normpath(os.path.join(project_path, path)).lower()
+                for path in relative_path_list]
+            # Limit the number of results to 500
+            if (len(result_list) > 500):
+                result_list = result_list[:500]
+            return result_list
+        except subprocess.CalledProcessError:
```

Comment:
Seems like here a `FileNotFoundError` (in case `fzf` is not installed)? So something like:

```suggestion
        except subprocess.CalledProcessError, FileNotFoundError:
```